tests/python_package_test/conftest.py

```diff
@@ -10,3 +12,10 @@ def rng():
 @pytest.fixture(scope="function")
 def rng_fixed_seed():
     return np.random.default_rng(seed=42)
+
+
+@pytest.fixture(scope="function")
+def missing_module_cffi(monkeypatch):
+    """Mock 'cffi' not being importable"""
+    monkeypatch.setattr(lightgbm.compat, "CFFI_INSTALLED", False)
+    monkeypatch.setattr(lightgbm.basic, "CFFI_INSTALLED", False)
```

Comment:
Came up with this based on https://docs.pytest.org/en/stable/reference/reference.html

I'm hoping that this could establish a pattern we re-use in other tests in future PRs.

It's not perfect (for example, if setting `CFFI_INSTALLED` is done incorrectly in `compat.py`, then this approach wouldn't catch that), but it's a lightweight and simple way to ensure we always cover code like the changes introduced in this PR.

Referenced https://docs.pytest.org/en/stable/reference/reference.html while working on this.

@jmoralez @borchero @StrikerRUS what do you think about this approach?

Reply:
This approach looks quite fragile. I think more complicated but right approach would be like one from the following ones: https://stackoverflow.com/a/51048604.

Reply:
That is a lot more complicated :/

I'll try it though, I do agree that it'd be a stronger test to go all the way into getting the import to literally raise an `ImportError`.

Reply:
@StrikerRUS do you want to try?

Reply:
Thanks a lot for trying this approach! Yes, I'd like to check some things but I don't want to block merging of this PR and next release.

Reply:
Alright thanks! I'll merge this after CI runs, then.

The Stack Overflow link you shared was excellent. In case you do try this, here are some other links I consulted:

* https://docs.python.org/3/reference/import.html
* https://docs.pytest.org/en/7.1.x/explanation/pythonpath.html
* https://github.com/jeertmans/pytest-missing-modules